cli/runtime/config/cli_discovery_sources_test.go

```diff
@@ -0,0 +1,433 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	configapi "github.com/vmware-tanzu/tanzu-framework/cli/runtime/apis/config/v1alpha1"
+)
+
+func TestGetCLIDiscoverySources(t *testing.T) {
+	// setup
+	func() {
+		LocalDirName = TestLocalDirName
+	}()
+	defer func() {
+		cleanupDir(LocalDirName)
+	}()
+	tests := []struct {
+		name   string
+		in     *configapi.ClientConfig
+		out    []configapi.PluginDiscovery
+		errStr string
+	}{
+		{
+			name: "success get all",
+			in: &configapi.ClientConfig{
+				ClientOptions: &configapi.ClientOptions{
+					CLI: &configapi.CLIOptions{
+						DiscoverySources: []configapi.PluginDiscovery{
+							{
+								GCP: &configapi.GCPDiscovery{
+									Name:         "test",
+									Bucket:       "updated-test-bucket",
+									ManifestPath: "test-manifest-path",
+								},
+								ContextType: configapi.CtxTypeTMC,
+							},
+						},
+					},
+				},
+			},
+			out: []configapi.PluginDiscovery{
+				{
+					GCP: &configapi.GCPDiscovery{
+						Name:         "test",
+						Bucket:       "updated-test-bucket",
+						ManifestPath: "test-manifest-path",
+					},
+					ContextType: configapi.CtxTypeTMC,
+				},
+			},
+		},
+	}
+	for _, spec := range tests {
+		t.Run(spec.name, func(t *testing.T) {
+			err := StoreClientConfig(spec.in)
+			if err != nil {
+				fmt.Printf("StoreClientConfigV2 errors: %v\n", err)
```

Comment:
nit: the 'V2' (in 9 places) looks odd, and is probably redundant (we have changed the implementation of StoreClientConfig, but it is not like we now start to test different implementations of StoreClientConfig, and need the suffix to differentiate among them)

Reply:
good catch. pushed the changes